packages/docusaurus/src/server/i18n.ts

```diff
@@ -85,20 +105,16 @@ export function localizePath({
   i18n: I18n;
   options?: {localizePath?: boolean};
 }): string {
-  const shouldLocalizePath: boolean =
-    typeof options.localizePath === 'undefined'
-      ? // By default, we don't localize the path of defaultLocale
-        i18n.currentLocale !== i18n.defaultLocale
-      : options.localizePath;
-
-  if (shouldLocalizePath) {
+  if (shouldLocalizePath({i18n, options})) {
     // FS paths need special care, for Windows support
+    const localePath =
+      i18n.localeConfigs[i18n.currentLocale].baseUrl || i18n.currentLocale;
```

Comment:
We prefer `??` over `||` nowadays, so if you change the above to `undefined`, you would also be able to use `??` here.